enhancements/ingress/contour-operator.md

```diff
@@ -0,0 +1,217 @@
+* Create tooling to simplify building, running, testing, etc. the operator.
+* Create tests that reduce bugs in the code and allow for continuous integration.
+* Integrate the operator with the OpenShift toolchain, i.e. openshift/release, required
+for productization.
+* Provide manifests for deploying the operator.
+* Add operator support to [must-gather](https://github.com/openshift/must-gather).
+* Add [telemetry support](https://github.com/openshift/cluster-monitoring-operator) for the operator and its operand(s).
+
+The following functionality is expected as part of the operator:
+
+* Introduce a CRD used by the operator to manage Contour and its dependencies. Any API(s) introduced by this enhancement
+will follow the Kubernetes API [deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/).
+__Note:__ Although the upstream operator will support projectcontour.io APIs, e.g.
+[HTTPProxy](https://projectcontour.io/docs/v1.4.0/httpproxy/), OpenShift will only support Service APIs.
+* Common fields in spec that define the configuration of Contour and its dependencies.
+* Common fields in spec that define how to deploy Contour, i.e. number of pod replicas.
+* Common fields in status that expose the current health & version information
+  of Contour.
+* The operator will be capable of observing other resources to perform basic sequencing
+required for OpenShift integration.
+
+An example can make this easier to understand, here is what a CRD instance for
+Contour might look like:
+
+```yaml
+apiVersion: projectcontour.io/v1alpha1
+kind: Contour
+metadata:
+  name: default
+  namespace: my-contour-namespace
+spec:
+  replicas: 2
+status:
+  availableReplicas: 2
+  conditions:
+  - lastTransitionTime: "2020-08-20T23:01:33Z"
+    reason: Valid
+    status: "True"
+    type: Admitted
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    status: "True"
+    type: Available
+  - lastTransitionTime: "2020-08-20T23:07:05Z"
+    message: The deployment has Available status condition set to True
+    reason: DeploymentAvailable
+    status: "False"
+    type: DeploymentDegraded
+```
+
+This manifest creates a Contour deployment that ensures 2 instances of Contour are always
+running in the cluster.  Additional `spec` fields may be introduced based on requirements.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a developer, I need the ability to add Service APIs support to OpenShift.
+
+#### Story 2
+
+As a cluster administrator, I need to use Service APIs to provide access to applications
+running in my OpenShift cluster.
+
+### Implementation Details [optional]
+
+TBD
+
+### Risks and Mitigations
+
+TBD
+
+## Design Details
+
+TBD
+
+### Open Questions [optional]
+
+1. Should the operator be created upstream to allow for community-based development and support? An
+[issue](https://github.com/projectcontour/contour/issues/2187) has been created to address this question
+with the Contour developer community.
```

Comment:
Open question I'd like added - answering "how would an openshift customer select an ingress controller of type contour" as a follow up enhancement would gate this leaving tech preview for me (i think)